dotrix_core/src/renderer.rs

```diff
@@ -13,6 +13,7 @@ pub use backend::{
     Bindings, PipelineBackend, Sampler, ShaderModule, StorageBuffer, TextureBuffer, UniformBuffer,
     VertexBuffer, WorkGroups,
 };
+pub use wgpu::{StorageTextureAccess, TextureFormat, TextureUsages};
```

Comment:
`TextureUsages` and `StorageTextureAccess ` I think we could do quite easily but there are 96 different `TextureFormats` which would make it a bit of work to map effectivly.

Reply:
We might be able to generalise it by channels and formats like this. 

`TextureFormat::Create().with_r().with_g().as_32float()`

`TextureFormat::Create().with_r().with_g().as_8snorm()`

Then match the result of that build to something in wgpu. I imagine the names will change but channels and data types should be generic.

Reply:
Yep, could be

Reply:
I've pushed a change which contains a mapping for `TextureUsages`. I will have a go at the rest in the morning, but if you have time please have a look at the general design as that is what I will be using for the rest too

Reply:
General design is fine. Use it.

Reply:
So I added it in for the other two, but I am not really happy with the `TextureFormat` one. I think I'd still just prefer the original enum. What are your comments on it? Use something like this/use a new enum/use the wgpu enum/something else?

Reply:
I have also missed our all of the compressed texture formats and the depth formats

Reply:
What about:

```rust
enum TextureFormat {
  R {
    channel_bits: 8,
    channel_format: ChannelFormat::UnormSrbg,
  }
  ...
```

There are only three combinations

R, Rg, Rgba what actually matters is the number of the channels not how we perceive them and there's no rgb perhaps because that would be 24bits on the gpu. What there are a lot of combinations of is ChannelFormat

Reply:
Ok so I changed it to an enum based design like I suggested please review when you have time :)

Reply:
I think we overcomplicate. What about to merge ChannelFormat and ChannelBits into one enum with only most common formats and then just made a wrapper around wgpu type with constructors:
```
pub struct TextureFormat {
    pub wgpu_texture_format: wgpu::TextureFormat,
}
impl TextureFormat {
  pub fn r(channel_format: ChannelFormat) {
    match channel_format {
       ...
    }
    pub fn rg(channel_format: ChannelFormat) { ... }
    pub fn rgba(channel_format: ChannelFormat) { ... }
  }
}
```
It will still provide an abstraction, but if necessary, direct usage of wgpu data. 
Maybe even ChannelFormat is too explicit. Maybe constructors will be enough:
```
fn rgba_u8();

fn rgba_u16();

fn rgba_u32();

etc...
```

Reply:
Ok I've set something like that up please have another look

Reply:
One small suggestion regarding `filterable`.  Otherwise it is OK.

Reply:
Ok Done